src/distance_sets.jl

```diff
@@ -131,6 +131,11 @@ function distance_to_set(::NormedEpigraphDistance{p}, v::AbstractVector{<:Real},
     x = v[1]
     y = v[2]
     z = v[3]
+
+    if x <= 0 && isapprox(y, 0, atol=1e-10) && z >= 0
+        return 0.0
```

Comment:
```suggestion
        return zero(eltype(v))
```

Reply:
this helps make the functions generic for different element types (Float64 / 32, BigFloat, etc)